webhook/main.py

```diff
@@ -1,39 +1,65 @@
-from enum import Enum
-from fastapi import FastAPI, Request, Header, HTTPException, Body
-from pydantic import BaseModel
-from typing import List, Optional, Dict, Union
+from fastapi import FastAPI, Header, HTTPException, status, Response, Body
+from fastapi.encoders import jsonable_encoder
+from google.cloud import logging, ndb
+from typing import Optional, Dict
+
+from models import VTAPI, APIKey, APIKeyEmail, UserEmail
+
+import json
+import logging as log
 
 app = FastAPI()
+client = ndb.Client()
+
+logger = logging.Client()
+logger.get_default_handler()
+logger.setup_logging()
 
 
 @app.get("/")
 async def root():
     return {"data": "Hello World"}
 
 
-class VTType(str, Enum):
-    file = 'file'
-    url = 'url'
-    domain = 'domain'
-    ip_address = 'ip_address'
+@app.post("/query-results/")
+async def send_query_results(request: VTAPI,
+                             x_appengine_inbound_appid: Optional[str] = Header(None)):
+    if x_appengine_inbound_appid != 'virustotal-step-2020':
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN, detail="Access forbidden")
+
+    email = None
+    with client.context():
+        email = ndb.Key("UserEmail", request.api_key).get()
 
+    if email is None:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN, detail="API Key is not valid")
+    
+    log.info("Send an email to %s.\nBody:\n%s".format(email, json.dumps(jsonable_encoder(request.data))))
 
-class VTData(BaseModel):
-    attributes: Dict
-    id: str
-    links: Dict
-    type: VTType
 
+@app.post("/email-address/")
+async def set_email(response: Response, content: APIKeyEmail):
+    # TODO Check why documentation is not generating correct response code_status.
```

Comment:
Do you still need this TODO?

Reply:
Yeah, I have no idea how to make docs recognise that there can be code 201. It's seing only 200 and 422 for now.